src/operun/crm/ldap/ldap_sync.py

```diff
@@ -0,0 +1,289 @@
+            name='operun.crm.ldap_server_uri')
+        ldap_service_user = api.portal.get_registry_record(
+            name='operun.crm.ldap_service_user')
+        ldap_service_pass = api.portal.get_registry_record(
+            name='operun.crm.ldap_service_pass')
+        if all([sync_to_ldap,
+                ldap_server_uri,
+                ldap_service_user,
+                ldap_service_pass]):
+            try:
+                logger.info(_('Contacting LDAP server...'))
+                self.connection = ldap.initialize(ldap_server_uri)
+                self.connection.simple_bind_s(
+                    ldap_service_user, ldap_service_pass)
+            except ldap.LDAPError:
+                logger.info(
+                    _('Could not connect to {0}'.format(ldap_server_uri))
+                )
+            else:
+                logger.info(_('Connected to {0}'.format(ldap_server_uri)))
+                return self.connection
+        else:
+            logger.info(_('Check LDAP configuration under settings.'))
+
+    def unbind(self):
+        """
+        Unbind connection.
+        """
+        try:
+            self.connection.unbind()
+        except AttributeError:
+            logger.info(_('No connection!'))
+        else:
+            logger.info(_('Disconnected from LDAP server...'))
+
+    # Common Utils
+
+    def list_to_dict(self, list_of_items):
+        """
+        Converts list mappings into dict.
+        """
+        return dict(item.split('|') for item in list_of_items)
+
+    def types_to_sync(self):
+        """
+        Return list of Content-Types to sync.
+        """
+        types = ['Account', 'Contact']
+        return types
+
+    def convert_to_object(self, obj):
+        """
+        Converts object if object has getObject() attribute.
+        """
+        if hasattr(obj, 'getObject'):
+            obj = obj.getObject()
+        elif hasattr(obj, 'object'):
+            obj = obj.object
+        return obj
+
+    def generate_mod_attrs(self, obj):
+        """
+        Generate list of attributes and values for use in LDAP.
+        """
+        mod_attrs = []
+        obj = self.convert_to_object(obj)
+        content_type = obj.Type()
+        if content_type in self.types_to_sync():
+            for field in self.get_fields_to_update(content_type):
+                mod_attrs.append(
+                    (self.get_mapped_field(content_type, field),
+                     str(getattr(obj, field)))
+                )
+        return mod_attrs
+
+    def generate_create_mod_attrs(self, obj):
+        """
+        Generates a mod_attrs list with objectClass and UID.
+        Used in LDAP create object mode.
+        """
+        mod_attrs = self.generate_mod_attrs(obj)
+        content_type = obj.Type()
+        object_uid = obj.UID()
+        class_list = {
+            'Account': 'posixGroup',
+            'Contact': 'inetOrgPerson',
+        }
```

Comment:
This should come from a separate method. Then we're able to change it to config later